observability/observability.go

```diff
@@ -116,6 +116,14 @@ func RecordTraceExporterMetrics(ctx context.Context, receivedSpans int, droppedS
 	stats.Record(ctx, mExporterReceivedSpans.M(int64(receivedSpans)), mExporterDroppedSpans.M(int64(droppedSpans)))
 }
 
+// MakeComponentName constructs the name of the component that should be used as a tag/metrics label.
```

Comment:
It is worth clarifying that a component currently means either a receiver or exporter (although it can be extended in the future to other components, e.g. processors or extensions).

Also, perhaps use `MakeComponentObservableName` as this function name instead to make it clear that this is the name of the component from the observability perspective? (as opposed to for example full component name in the config).

Reply:
Updated the comment.

For the name I think it is fine right now because the function is already in "observability" package.

Reply:
Sounds good.